osrparse/utils.py

```diff
@@ -193,3 +195,66 @@ class LifeBarState:
     """
     time: int
     life: float
+
+class Statistics(TypedDict, total=False):
```

Comment:
Lazer only types it as a [`Dictionary<HitResult, int>`](https://github.com/ppy/osu/blob/790f863e0654fd563b57ab699d6be86895e756ab/osu.Game/Scoring/Legacy/LegacyReplaySoloScoreInfo.cs#L34). Since all possible HitResult values are stored in rulesets (`Ruleset.GetValidHitResults()`) I could make a separate class for each ruleset? 

As for always present fields across all rulesets only `miss` and `great` are shared between all of them.